### ISSUE TITLE
[upgrade test failure] PodPreset should create a pod preset; PodPreset should not modify the pod on conflict

### ISSUE THREAD

**Issue by ericchiang** [kubernetes/kubernetes#52587]
cc @kubernetes/sig-service-catalog-test-failures 
cc @kubernetes/kubernetes-release-managers 

Both tests failing on our upgrade cluster test suite

https://k8s-testgrid.appspot.com/master-upgrade#gce-1.7-master-upgrade-cluster-parallel

Example failures

https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/ci-kubernetes-e2e-gce-new-master-upgrade-cluster-parallel/568

Looks like an API group not being enabled?

```
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/podpreset.go:151
Expected error:
    <*errors.StatusError | 0xc420018e00>: {
        ErrStatus: {
            TypeMeta: {Kind: "", APIVersion: ""},
            ListMeta: {SelfLink: "", ResourceVersion: ""},
            Status: "Failure",
            Message: "the server could not find the requested resource (post podpresets.settings.k8s.io)",
            Reason: "NotFound",
            Details: {
                Name: "",
                Group: "settings.k8s.io",
                Kind: "podpresets",
                UID: "",
                Causes: [
                    {
                        Type: "UnexpectedServerResponse",
                        Message: "404 page not found",
                        Field: "",
                    },
                ],
                RetryAfterSeconds: 0,
            },
            Code: 404,
        },
    }
    the server could not find the requested resource (post podpresets.settings.k8s.io)
not to have occurred
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/podpreset.go:75
```

**Comment by liggitt** [comment]
Pod preset alpha group is no longer enabled by default. The e2e test is behind a feature tag.

**Comment by liggitt** [comment]
Feature flag added to e2e test in 1.7 in https://github.com/kubernetes/kubernetes/pull/52615

**Comment by ericchiang** [comment]
PR merged into the 1.7 release branch. Closing.
